File: Server/Models/Entities/Scope.cs
using System.ComponentModel.DataAnnotations;

namespace Authentication.Models.Entities;

public class Scope
{
    [Key] public int ScopeId { get; set; }

    public string ScopeName { get; set; }
    public int ScopeType { get; set; }
    public string Slug { get; set; }
    public Guid? CreatedBy { get; set; }
    public int? ImpactLevelId { get; set; }
    public int? CategoryId { get; set; }
}

public record ScopeCreateDto(
    string ScopeName,
    int ScopeType,
    string Slug,
    Guid CreatedBy,
    int? ImpactLevelId,
    int? CategoryId
);

File: website/models/scope.ts
import {z} from "zod";

export const ScopeSchema = z.object({
    scope_id: z.number(),
    scope_name: z.string(),
    scope_type: z.number(),
    slug: z.string(),
    created_by: z.string().uuid().optional(),
    impact_level_id: z.number().optional(),
    category_id: z.number().optional(),
});

export type Scope = z.infer<typeof ScopeSchema>;